homeassistant/components/habitica/button.py

```diff
@@ -259,6 +262,38 @@ async def async_setup_entry(
     """Set up buttons from a config entry."""
 
     coordinator = entry.runtime_data
+    listener: Callable[[], None] | None = None
+    not_setup: set[HabiticaButtonEntityDescription] = set(
+        HEALER_SKILLS + MAGE_SKILLS + ROGUE_SKILLS + WARRIOR_SKILLS
+    )
```

Comment:
You can only have 1 setup at a time right?

Reply:
?

Reply:
You only have one class right?

Reply:
yes, checking for the players class and selection which skills get loaded happens in the callback

```python
            if (
                coordinator.data.user["stats"]["lvl"] >= 10
                and coordinator.data.user["flags"]["classSelected"]
                and not coordinator.data.user["preferences"]["disableClasses"]
                and description.class_needed == coordinator.data.user["stats"]["class"]
            ):
                buttons.append(HabiticaButton(coordinator, description))
```

Reply:
But why do we create the full list then? Can't we compare against the chosen class and store for what entities the class has been added?